### DISCUSSION TITLE
User stories

### DISCUSSION THREAD

**Discussion by accuser** [swe-delivery/swe-tube#6] [Ideas]
I've used student in place of apprentice and teacher in place of coach / instructor.

1. As a student, I want to be able to find resources related to a specific programming language or topic, so that I can quickly access the materials that are relevant to me. 
2. As a teacher, I want to be able to curate a list of resources that I can share with my students, so that I can ensure that they have access to high-quality materials that will help them learn. 
4. As a student, I want to be able to submit a new resource that I think would be useful for my peers, so that I can contribute to the community and help others learn.
5. As a teacher, I want to be able to moderate student submissions, so that I can ensure that the resources that are added to the site are appropriate and relevant.
6. As a student, I want to be able to browse resources by category, so that I can discover new materials related to topics that interest me.
7. As a student, I want to be able to search for resources by keyword, so that I can quickly find the materials that I need.
8. As a teacher, I want to be able to create custom categories and tags for the resources on the site, so that I can organize them in a way that makes sense for my students.
9. As a user, I want to be able to easily share links to resources on social media, so that I can help spread the word about SWE-Tube and the resources that it provides.
10. As a student, I want to be able to see which resources have been recommended by my peers or by my teacher, so that I can quickly find high-quality materials that have been vetted by others.
11. As a user, I want to be able to view the site on a mobile device, so that I can access the resources even when I'm not at my computer.

**Comment by nimonian** [comment]
For the bootcamp iteration, should we aim to implement the "student" stories only? I'm just thinking that avoiding authentication will make it harder to have different roles, and roles feels like they would be hard to accomplish based only on what we learn in bootcamp.

**Comment by Jaardy** [comment]
For Students:

-   As a student, I want to be able to find resources related to a specific programming language or topic, so that I can quickly access the materials that are relevant to me.
-   As a student, I want to be able to submit a new resource that I think would be useful for my peers, so that I can contribute to the community and help others learn.
-   As a student, I want to be able to browse resources by category, so that I can discover new materials related to topics that interest me.
-   As a student, I want to be able to search for resources by keyword, so that I can quickly find the materials that I need.
-   As a student, I want to be able to see which resources have been recommended by my peers or by my teacher, so that I can quickly find high-quality materials that have been vetted by others.

For Teachers:

-   As a teacher, I want to be able to curate a list of resources that I can share with my students, so that I can ensure that they have access to high-quality materials that will help them learn.
-   As a teacher, I want to be able to moderate student submissions, so that I can ensure that the resources that are added to the site are appropriate and relevant.
-   As a teacher, I want to be able to create custom categories and tags for the resources on the site, so that I can organize them in a way that makes sense for my students.

For Users (including both students and teachers):

-   As a user, I want to be able to easily share links to resources on social media, so that I can help spread the word about SWE-Tube and the resources that it provides.
-   As a user, I want to be able to view the site on a mobile device, so that I can access the resources even when I'm not at my computer.

**Reply by Jaardy** [reply]
Same idea of learning by osmosis.  Target Student stories, but give glimpses of the big picture.

**Reply by accuser** [reply]
Remove:

- As a student, I want to be able to submit a new resource that I think would be useful for my peers, so that I can contribute to the community and help others learn.

And we can implement these stories without the need for auth in the vanilla / standalone version.

**Reply by nimonian** [reply]
That's perfect :raised_hands:
